src/ResourceManagement/Azure.Fluent/Fluent.Tests/Network/ApplicationGateway/PublicComplex.cs

```diff
@@ -0,0 +1,243 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.Txt in the project root for license information.
+
+using Azure.Tests.Common;
+using Microsoft.Azure.Management.Network.Fluent;
+using Microsoft.Azure.Management.Network.Fluent.Models;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Xunit;
+
+namespace Azure.Tests.Network.ApplicationGateway
+{
+    /// <summary>
+    /// Internal complex app gateway test.
+    /// </summary>
+    public class PublicComplex : TestTemplate<IApplicationGateway, IApplicationGateways>
+    {
+        private List<IPublicIpAddress> testPips;
+        private INetwork network;
+
+        public PublicComplex(IPublicIpAddresses pips)
+        {
+            testPips = new List<IPublicIpAddress>(ApplicationGatewayHelper.EnsurePIPs(pips));
+        }
+
+        public override void Print(IApplicationGateway resource)
+        {
+            ApplicationGatewayHelper.PrintAppGateway(resource);
+        }
+
+        public override IApplicationGateway CreateResource(IApplicationGateways resources)
+        {
+            //Thread creationThread = new Thread(() =>
+            //{
+            //    Thread.CurrentThread.IsBackground = true;
+            //    // Create an application gateway
```

Comment:
ok. (commented out because this trick should work, but for an inexplicable reason it doesn't in this specific test. Anyway....)